packages/aws-cdk-lib/aws-rds/lib/instance.ts

```diff
@@ -1366,7 +1366,7 @@ export class DatabaseInstanceReadReplica extends DatabaseInstanceNew implements
     this.instanceIdentifier = instance.ref;
     this.dbInstanceEndpointAddress = instance.attrEndpointAddress;
     this.dbInstanceEndpointPort = instance.attrEndpointPort;
-    this.instanceResourceId = instance.attrDbInstanceArn;
```

Comment:
I am afraid that this can be a breaking change to customers. I know that we assigned a wrong value, but may be there are some customers that use this property to get the DB Instance Arn and use it some how. So, my recommendation is to deprecate this property, and to add a new property for the instance resource id, and use it in grant method, and also check where else we use the current property.

Reply:
@moelasmar Thanks for your review comment. Would creating a feature flag (example https://github.com/aws/aws-cdk/pull/29794) be the recommended approach instead of adding a new property? Please advise.

Reply:
I think the feature flag here will not fix the problem, as the flag default value will be false (to avoid breaking customers), so the policy will still referring to the wrong replica instance id. So, I do not prefer feature flag, it add a lot of confusion on us when we write code, so we can use this property in similar cases assuming that we are using the correct property, but because the feature flag is disabled, this may cause more issues for customers.

I prefer to go with deprecation, and adding new property.

Reply:
@moelasmar Thanks for your review comment. I have added a commit to:
- Deprecate `instanceResourceId` property and add new `instanceResourceIdV2` property in `DatabaseInstanceReadReplica`.
- Added a copy of `grantConnect()` in `DatabaseInstanceReadReplica` that makes use of `instanceResourceIdV2`.

Please review.

Thanks,
Ashish

Reply:
Thanks @ashishdhingra for resolving my comment, and sorry for late response. 

I actually, did a deeper check, and I found that I missed that the property `instanceResourceId` is actually exist in the Interface `IDatabaseInstance` so, adding a new property will actually break the idea of that interface, and also deprecating this property is not a good idea.

So, I think it is better to make this change under feature flag, and to enable the feature flag by default, and the customers who want to keep the current behaviour, they can disable the flag.

Reply:
I pushed some updates to this change to add the feature flag. Could you please take a look.

Reply:
Looks good reviewing at high level. Needs review from the maintainer.